DMX.Core.Api/Services/Foundations/LabCommands/LabCommandService.Validations.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using DMX.Core.Api.Models.Foundations.LabCommands;
+using DMX.Core.Api.Models.Foundations.LabCommands.Exceptions;
+using CommandType = DMX.Core.Api.Models.Foundations.LabCommands.CommandType;
+
+namespace DMX.Core.Api.Services.Foundations.LabCommands
+{
+    public partial class LabCommandService
+    {
+        private void ValidateLabCommandOnAdd(LabCommand labCommand)
+        {
+            ValidateLabCommandIsNotNull(labCommand);
+
+            Validate(
+                (Rule: IsInvalid(labCommand.Id), Parameter: nameof(LabCommand.Id)),
+                (Rule: IsInvalid(labCommand.LabId), Parameter: nameof(LabCommand.LabId)),
+                (Rule: IsInvalid(labCommand.Arguments), Parameter: nameof(LabCommand.Arguments)),
+                (Rule: IsInvalid(labCommand.Notes), Parameter: nameof(LabCommand.Notes)),
+                (Rule: IsInvalid(labCommand.Status), Parameter: nameof(LabCommand.Status)),
+                (Rule: IsInvalid(labCommand.Type), Parameter: nameof(LabCommand.Type)));
```

Comment:
We must validate few things here:
1. Created Date & Updated Date are the same value
2. Created Date is recent (you will need to inject `DateTimeBroker` for that.)
Here's some examples:
[Code example](https://github.com/hassanhabib/Taarafo.Core/blob/master/Taarafo.Core/Services/Foundations/Posts/PostService.Validations.cs#L25)

[Vido Example](https://www.youtube.com/watch?v=0vEIY1kzduw&ab_channel=HassanHabib)